src/System.Net.Security/src/System/Net/NegotiateStreamPal.Unix.cs

```diff
@@ -134,7 +213,32 @@ internal static Exception CreateExceptionFromError(SecurityStatusPal statusCode)
             out int newOffset,
             uint sequenceNumber)
         {
-            throw new PlatformNotSupportedException();
+            if (offset < 0 || offset > (buffer == null ? 0 : buffer.Length))
```

Comment:
@shrutigarg we copied this from Windows code so make sure to fix it in NegotiateStreamPal.Windows.cs as well